sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -634,10 +634,6 @@ private static Query translateGqlQuery(String gql, Datastore datastore, String n
     public void validate(PipelineOptions options) {
       checkNotNull(getProjectId(), "projectId");
 
-      if (getProjectId().isAccessible() && getProjectId().get() == null) {
```

Comment:
What if the person passed in a StaticValueProvider containing a null?

Reply:
I'm not sure we need to be this diligent about nullness checking, but okay, reverted this part.